public/js/maho-captcha.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Maho
+ *
+ * @package     Maho_Captcha
+ * @copyright   Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+const MahoCaptcha = {
+    loading: null,
+    altchaWidget: null,
+    altchaState: null,
+    frontendSelectors: '',
+    onVerifiedCallback: null,
+
+    async setup(config) {
+        this.altchaWidget = document.querySelector('altcha-widget');
+        this.frontendSelectors = config.frontendSelectors ?? '';
+        this.loading = config.loading ?? '';
+
+        if (document.readyState === 'loading') {
+            document.addEventListener('DOMContentLoaded', this.initForms.bind(this));
+        } else {
+            this.initForms();
+        }
+    },
+
+    initForms() {
+        for (const formEl of document.querySelectorAll(this.frontendSelectors)) {
+            formEl.addEventListener('focusin', this.loadAltchaScripts.bind(this), true);
+            formEl.addEventListener('submit', this.onFormSubmit.bind(this), true);
+        }
+    },
+
+    async loadAltchaScripts() {
+        // Load Altcha JS
+        await Promise.all([
```

Comment:
The removed if check here makes it so a new `altcha.min.js` is loaded every time you focus in a form:
```js
if (typeof customElements.get('altcha-widget') !== 'function') {
```

![image](https://github.com/user-attachments/assets/2b9c5828-e373-4ce4-b474-25ef640aa540)